go/tools/gazelle/README.md

```diff
@@ -1,35 +1,274 @@
-# Gazelle BUILD file generator
+# Gazelle build file generator
+
+Gazelle is a build file generator for Go projects. It can create new
+BUILD.bazel files for a project that follows "go build" conventions, and it
+can update existing build files to include new files and options. Gazelle can
+be invoked directly in a project workspace, or it can be run on an external
+repository during the build as part of the `go_repository` rule.
+
+*Gazelle is under active development. Its interface and the rules it generates
+may change.*
+
+## Contents
+
+* [Setup](#setup)
+  * [Running Gazelle with Bazel](#running-gazelle-with-bazel)
+  * [Running Gazelle separately](#running-gazelle-separately)
+* [Usage](#usage)
+  * [Command line](#command-line)
+  * [Bazel rule](#bazel-rule)
+  * [Directives](#directives)
 
 ## Setup
 
-For local use, in `$GOPATH/src/github.com/bazelbuild/rules_go/go/tools/gazelle/gazelle`
-run `go install`
+### Running Gazelle with Bazel
+
+To use Gazelle in a new project, add the following to the BUILD or BUILD.bazel
+file in the root directory of your repository:
+
+```bzl
+load("@io_bazel_rules_go//go:def.bzl", "gazelle", "go_prefix")
+
+go_prefix("github.com/my/project")
+
+gazelle(name = "gazelle")
+```
+
+Replace the string in `go_prefix` with the portion of your import path that
+corresponds to your repository. See
+[`go_prefix` documentation](../../../README.md#go_prefix).
+
+After adding those rules, run the command below:
+
+```
+bazel run //:gazelle
+```
+
+This will generate new BUILD.bazel files for your project. You can run the same
+command in the future to update existing BUILD.bazel files to include new source
+files or options.
+
+### Running Gazelle separately
+
+If you have a Go SDK installed, you can install Gazelle in your `GOPATH` with
+the command below:
+
+```
+go get -u github.com/bazelbuild/rules_go/go/tools/gazelle/gazelle
+```
+
+Make sure to re-run this command to upgrade Gazelle whenever you upgrade
+rules_go in your repository.
+
+To generate BUILD.bazel files in a new project, run the command below, replacing
+the prefix with the portion of your import path that corresponds to your
+repository.
+
+```
+gazelle -go_prefix github.com/my/project
+```
+
+The prefix only needs to be specified the first time you run Gazelle. To update
+existing BUILD.bazel files, you can just run `gazelle` without arguments.
 
 ## Usage
 
-  gazelle
-  
-Which will fix all build files in the current directory plus subdirectories.
+### Command line
+
+```
+gazelle <command> [flags...] [package-dirs...]
+```
+
+The first argument to Gazelle may be one of the commands below. If no command
+is specified, `update` is assumed.
+
+<table class="table table-condensed table-bordered table-params">
```

Comment:
I looked into this, but I don't think I can use it. Markdown tables can't include multi-line content like paragraphs (used in the flags table) or lists. Each row of the table has to fit on a single (long) line, so it will be even less readable than the HTML.